app/scripts/models/password_strength/password_strength_balloon.js

```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Manage state needed to display the password strength balloon, aka, Design F
+ *
+ * @export
+ * @class PasswordStrengthBalloonModel
+ * @extends {Model}
+ */
+import { assign } from 'underscore';
+import AuthErrors from '../../lib/auth-errors';
+import { Model } from 'backbone';
+import { PASSWORD_MIN_LENGTH } from '../../lib/constants';
+
+// Ban passwords that start with firefox, fxaccounts, lockbox, addons, sumo
+const BANNED_SERVICE_REGEXP = /^(firefox|fxaccounts|lockbox|addons|sumo).*$/;
+const BANNED_URL_REGEXP = /^(?:firefox|mozilla)\.(?:com|org)$/;
+
+export default class PasswordStrengthBalloonModel extends Model {
+  constructor (attrs = {}, config = {}) {
+    const attrsWithDefaults = assign({
+      hasEnteredPassword: false,
+      isCommon: false,
+      isSameAsEmail: false,
+      isTooShort: true,
+      isValid: false,
+    }, attrs);
+    super(attrsWithDefaults, config);
+
+    // Asynchronously load the common password list as soon as the model is created.
+    this._getCommonPasswordList();
+  }
+
+  _getCommonPasswordList () {
+    return import(/* webpackChunkName: "fxa-common-password-list" */ 'fxa-common-password-list');
+  }
+
+  /**
+   * Calculate model values for `password`
+   *
+   * @param {String} password
+   * @returns {Promise} resolves when complete
+   */
+  updateForPassword (password) {
+    return this._getCommonPasswordList().then((commonPasswordList => {
+      // user the lowercase password for comparison.
```

Comment:
Typo: `use`?

Reply:
It is a little strange to me to lower case the password since passwords are case sensitive. Do we want to treat `PaSSwoRD` the same as `password`, even though one is harder to guess than the other?